sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py

```diff
@@ -320,7 +323,7 @@ async def send(
                     request=request,
                     internal_response=result,
                     block_size=self.connection_config.data_block_size,
-                    decompress=not auto_decompress,
+                    decompress=decompress if decompress is not None else not auto_decompress,
```

Comment:
If customer sets `decompress` directly, we should respect that value, otherwise keep pre-existing behavior 😄